### PR TITLE
feat(plugins): add less import plugin

### PR DESCRIPTION
**Pull request by frenautvh**

### CODE REVIEW

File: src/plugins/less-import.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const { createFilter } = require('rollup-pluginutils');
+const fs = require('fs');
+const MagicString = require('magic-string');
+const path = require('path');
+
+module.exports = (opts = {}) => {
```
Comment (FredericEspiau) [inline src/plugins/less-import.js]:
```suggestion
module.exports = (opts = 
  {
    include = '**/*.less',
    path = ['./node_modules'],
  }
) => {
```

File: src/plugins/less-import.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const include = opts.include || '**/*.less';
+  const { exclude } = opts;
+  const filter = createFilter(include, exclude);
+  const filtering = opts.filtering !== false;
```
Comment (FredericEspiau) [inline src/plugins/less-import.js]:
```suggestion
  const filtering = !!opts.filtering;
```

not sure though

File: src/plugins/less-import.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const paths = opts.paths || ['./node_modules'];
+  return {
+    name: 'less-import',
+    transform(code, id) {
+      if (filtering && !filter(id)) return null;
+      const magicString = new MagicString(code);
+      const reg = /(@import\s*['"]+)(~[^'"]+)(['"]+\s*;)/g;
```
Comment (FredericEspiau) [inline src/plugins/less-import.js]: Regex for what ? Could you please name it better ?

File: src/plugins/less-import.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      let match = reg.exec(code);
+
+      while (match) {
```
Comment (FredericEspiau) [inline src/plugins/less-import.js]:
```suggestion
      do {
        match = reg.exec(code);
        ...
      } while (match);
```

File: src/plugins/less-import.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        let lessImport = match[2];
+        let i;
+
+        for (i = 0; i < paths.length; i += 1) {
+          const absolutePath = path.resolve(path.join(paths[i], lessImport.replace(/^~/, '')));
+          if (fs.existsSync(absolutePath)) {
+            lessImport = absolutePath;
+            break;
+          }
+          tries.push(absolutePath);
```
Comment (FredericEspiau) [inline src/plugins/less-import.js]: It looks like this could be in a `map`:

```js
paths.map(path => ...
```

File: src/index.js [hunk 6]
```diff
@@ -91,17 +95,18 @@ const iife = (opts) => {
...
-  umd: (opts = {}) => umd(merge(opts, globalOpts)),
+const config = (globalOpts = {}, pluginsOpts = {}) => ({
+  cjs: (opts = {}) => cjs(merge(opts, globalOpts), pluginsOpts),
+  es: (opts = {}) => es(merge(opts, globalOpts), pluginsOpts),
+  iife: (opts = {}) => iife(merge(opts, globalOpts), pluginsOpts),
+  umd: (opts = {}) => umd(merge(opts, globalOpts), pluginsOpts),
 });
 
 config.plugins = {
+  lessImport,
```
Comment (antleblanc) [inline src/index.js]: Could you also update `README.md` file in order to provide some documentation?

File: src/plugins/less-import.js (hunk 4 above)
Comment (frenautvh) [reply src/plugins/less-import.js]: then i would have to add an ```if (match) { ... }``` !

**Pull request closed without merging**